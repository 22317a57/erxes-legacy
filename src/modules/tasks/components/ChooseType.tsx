import Button from 'modules/common/components/Button';
import FormGroup from 'modules/common/components/form/Group';
import ControlLabel from 'modules/common/components/form/Label';
import Icon from 'modules/common/components/Icon';
import Tip from 'modules/common/components/Tip';
import { __ } from 'modules/common/utils';
import { ITaskType } from 'modules/settings/taskType/types';
import React from 'react';
import { Link } from 'react-router-dom';
import { TaskTypes, TypeIcon } from '../styles';

type Props = {
  types: ITaskType[];
  currentType?: string;
  onChange: (name: any, value: string) => void;
};

export default class ChooseType extends React.Component<Props> {
  renderTaskType() {
    const { types, onChange, currentType } = this.props;

    if (types.length === 0) {
      return (
        <Link to="/settings/task-types">
          <Button btnStyle="primary" size="small" icon="add">
            {__('Create task type')}
          </Button>
        </Link>
      );
    }

    return types.map(type => (
      <Tip key={type._id} text={type.name} placement="bottom">
        <TypeIcon
          className={currentType === type._id ? 'active' : ''}
          onClick={onChange.bind(this, 'typeId', type._id)}
        >
          <Icon icon={type.icon} size={15} />
        </TypeIcon>
      </Tip>
    ));
  }

  render() {
    return (
      <FormGroup>
        <ControlLabel>Type</ControlLabel>
        <TaskTypes>{this.renderTaskType()}</TaskTypes>
      </FormGroup>
    );
  }
}
