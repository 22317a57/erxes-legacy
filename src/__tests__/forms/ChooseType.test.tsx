import { mount, shallow } from 'enzyme';
import ChooseType from 'modules/forms/components/step/ChooseType';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

describe('ChooseType component', () => {
  const defaultProps = {
    type: 'string',
    onChange: (name: 'type', value: string) => null,
    color: 'pink',
    theme: 'default'
  };

  test('renders shallow successfully', () => {
    shallow(<ChooseType {...defaultProps} />);
  });

  test('renders mount with default props', () => {
    const control = mount(<ChooseType {...defaultProps} />);
    const props = control.props();

    expect(props).toMatchObject(defaultProps);
  });

  test('snapshot matches', () => {
    const rendered = renderer.create(<ChooseType {...defaultProps} />).toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
