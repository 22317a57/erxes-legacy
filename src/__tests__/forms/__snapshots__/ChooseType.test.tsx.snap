// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseType component snapshot matches 1`] = `
<div
  className="sc-fAJaQT gLizFN"
>
  <div
    className="sc-jdeSqf csfTEc"
  >
    <div
      className="sc-jKVCRD eDSGVr"
    >
      <label
        className="sc-ebFjAB bHuvQk"
      >
        Choose a flow type
      </label>
    </div>
    <div
      className="sc-fdQOMr iDPOke"
    >
      <div
        className="sc-jotlie becxzt sc-hSdWYo iFcZeJ"
        onClick={[Function]}
        selected={false}
      >
        <i
          className="icon-speech-bubble-3 sc-bwzfXH hpxbGw"
          icon="speech-bubble-3"
        />
        <span>
          ShoutBox
        </span>
      </div>
      <div
        className="sc-jotlie becxzt sc-hSdWYo iFcZeJ"
        onClick={[Function]}
        selected={false}
      >
        <i
          className="icon-expand sc-bwzfXH hpxbGw"
          icon="expand"
        />
        <span>
          Popup
        </span>
      </div>
    </div>
    <div
      className="sc-fdQOMr iDPOke"
    >
      <div
        className="sc-jotlie becxzt sc-hSdWYo iFcZeJ"
        onClick={[Function]}
        selected={false}
      >
        <i
          className="icon-computer sc-bwzfXH hpxbGw"
          icon="computer"
        />
        <span>
          Embedded
        </span>
      </div>
      <div
        className="sc-jotlie becxzt sc-hSdWYo iFcZeJ"
        onClick={[Function]}
        selected={false}
      >
        <i
          className="icon-downarrow sc-bwzfXH hpxbGw"
          icon="downarrow"
        />
        <span>
          Dropdown
        </span>
      </div>
    </div>
    <div
      className="sc-fdQOMr iDPOke"
    >
      <div
        className="sc-jotlie becxzt sc-hSdWYo iFcZeJ"
        onClick={[Function]}
        selected={false}
      >
        <i
          className="icon-logout-2 sc-bwzfXH hpxbGw"
          icon="logout-2"
        />
        <span>
          Slide-in Left
        </span>
      </div>
      <div
        className="sc-jotlie becxzt sc-hSdWYo iFcZeJ"
        onClick={[Function]}
        selected={false}
      >
        <i
          className="icon-logout-1 sc-bwzfXH hpxbGw"
          icon="logout-1"
        />
        <span>
          Slide-in Right
        </span>
      </div>
    </div>
  </div>
  <div
    className="sc-cBrjTV iWStWz"
  >
    <div
      className="sc-aewfc dsStAJ"
    >
      <div
        className="sc-isBZXS htStnM"
      >
        <div
          className="sc-OxbzP jocSpk"
          style={
            Object {
              "backgroundColor": "default",
            }
          }
        >
          <div />
        </div>
        <div
          className="sc-lnrBVv jitmGX"
        >
          <div
            className="sc-fvLVrH kjchyg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
